base/boot.jl

```diff
@@ -269,9 +269,9 @@ Void() = nothing
 
 immutable VecElement{T}
     value::T
-    VecElement{T}(value::T) where T = new(value) # disable converting constructor in Core
+    VecElement{T}(value::T) where {T} = new(value) # disable converting constructor in Core
 end
-VecElement(arg::T) where T = VecElement{T}(arg)
+VecElement(arg::T) where {T} = VecElement{T}(arg)
```

Comment:
For simple cases like this, I find the older syntax `VecElement{T}(arg::T) = VecElement{T}(arg)` to be much clearer than `where {T}`.   Why do we need to use `where` here?